File: Data Structures & Algorithms/contains-duplicate-ii/submission-4.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        # num_map = {}
        # for i in range(len(nums)):
        #     if nums[i] in num_map.keys():
        #         if abs(i - num_map[nums[i]]) <= k:
        #             return True
        #     num_map[nums[i]] = i
        # return False
        window = set()
        L = 0
        for R in range(len(nums)):
            if R - L > k:
                window.remove(nums[L])
                L += 1
            if nums[R] in window:
                return True
            window.add(nums[R])
        return False



        

File: Data Structures & Algorithms/duplicate-integer/submission-4.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        //boolean res = false;
        int j = 0;
        while (j < nums.length){
            for (int i = 0; i < nums.length; i++){
                if ((i != j) && (nums[i] == nums[j])){
                    return true;
                }
            }
            j++;
        }
        return false;
    }
}
